### ISSUE TITLE
manual_abs_diff suggested fix doesn't work for borrowed values

### ISSUE THREAD

**Issue by CattleProdigy** [rust-lang/rust-clippy#15254]
### Summary

The existing message is still helpful, obviously, but it would be great if the lint could detect the case that one of the arguments is a reference and implements `Copy`. It could then modify the suggested fix to dereference.

### Lint Name

manual_abs_diff

### Reproducer

I tried this code:

```rust
fn diff_check(a: &usize, b: &usize) -> usize {
    if a < b {
        b - a
    } else {
        a - b
    }
}
```

I saw this happen:

```
warning: failed to automatically apply fixes suggested by rustc to crate `observations`

after fixes were automatically applied the compiler reported errors within these files:

  * libraries/observations/src/storage/index.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0308]: mismatched types
    --> libraries/observations/src/storage/index.rs:134:16
     |
134  |     b.abs_diff(a)
     |       -------- ^ expected `usize`, found `&usize`
     |       |
     |       arguments to this method are incorrect
     |
note: method defined here
    --> /home/paul/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/num/mod.rs:1228:5
     |
1228 | /     uint_impl! {
1229 | |         Self = usize,
1230 | |         ActualT = u64,
1231 | |         SignedT = isize,
...    |
1245 | |         bound_condition = " on 64-bit targets",
1246 | |     }
     | |_____^
     = note: this error originates in the macro `uint_impl` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider dereferencing the borrow
     |
134  |     b.abs_diff(*a)
     |                +

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0308`.

```

I expected to see this happen:
A fix to be applied

### Version

```text
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: x86_64-unknown-linux-gnu
release: 1.88.0
LLVM version: 20.1.5
```

**Comment by CattleProdigy** [comment]
I've got a fix ready: https://github.com/CattleProdigy/rust-clippy/commit/8575bfc16a582b475396c899d564abcfdbefaae7
